core/src/main/scala/org/apache/spark/sql/execution/columnar/ExternalStoreUtils.scala

```diff
@@ -283,39 +283,47 @@ object ExternalStoreUtils extends Logging {
   def getConnectionProperties(session: Option[SparkSession], url: String, driver: String,
       dialect: JdbcDialect, poolProps: Map[String, String], connProps: Properties,
       executorConnProps: Properties, hikariCP: Boolean): ConnectionProperties = {
-    if (session.isDefined) {
-      val (user, password) = getCredentials(session)
-
-      if (!user.isEmpty && !password.isEmpty) {
-        def secureProps(props: Properties): Properties = {
-          props.setProperty(Attribute.USERNAME_ATTR, user)
-          props.setProperty(Attribute.PASSWORD_ATTR, password)
-          props
-        }
+    session match {
+      case Some(s) => getConnProps(session.get, url, driver, dialect, poolProps, connProps,
+        executorConnProps, hikariCP)
+      case None => ConnectionProperties(url, driver, dialect, poolProps, connProps,
+        executorConnProps, hikariCP)
+    }
+  }
 
-        // Hikari only take 'username'. So does Tomcat
-        def securePoolProps(props: Map[String, String]): Map[String, String] = {
-          props + (Attribute.USERNAME_ALT_ATTR.toLowerCase -> user) + (Attribute.PASSWORD_ATTR ->
-              password)
-        }
+  def getConnProps(session: SparkSession, url: String, driver: String, dialect: JdbcDialect,
+      poolProps: Map[String, String], connProps: Properties, executorConnProps: Properties,
+      hikariCP: Boolean):ConnectionProperties = {
+    val (user, password) = getCredentials(session)
 
-        ConnectionProperties(url, driver, dialect, securePoolProps(poolProps),
-          secureProps(connProps), secureProps(executorConnProps), hikariCP)
-      } else {
-        ConnectionProperties(url, driver, dialect, poolProps, connProps, executorConnProps,
-          hikariCP)
+    if (!user.isEmpty && !password.isEmpty) {
+      def secureProps(props: Properties): Properties = {
+        props.setProperty(Attribute.USERNAME_ATTR, user)
+        props.setProperty(Attribute.PASSWORD_ATTR, password)
+        props
       }
-    } else ConnectionProperties(url, driver, dialect, poolProps, connProps, executorConnProps,
-      hikariCP)
+
+      // Hikari only take 'username'. So does Tomcat
+      def securePoolProps(props: Map[String, String]): Map[String, String] = {
+        props + (Attribute.USERNAME_ALT_ATTR.toLowerCase -> user) + (Attribute.PASSWORD_ATTR ->
+            password)
+      }
```

Comment:
Are pool properties settings required now that we key by userName and each user already has a separate pool? I think this can be removed.